ui/StatusQ/src/StatusQ/Validators/AmountValidator.qml

```diff
@@ -0,0 +1,69 @@
+import StatusQ 0.1
+
+/*!
+  \qmltype AmountValidator
+  \inherits GenericValidator
+  \inqmlmodule StatusQ.Validators
+  \brief Validator validating amounts provided by the user.
+
+  The validator do following checks:
+
+  - marks empty input and consisting only of decimal point as Intermediate
+  - limits allowed char set - digits and (only when maxDecimalDigits is not 0)
+    two decimal point characters are available (".", ",")
+  - replaces entered decimal point to the one provied via decimalPoint property
+  - blocks attemps of entering more then one decimal point char
+  - limits number of integral part specified by maxIntegralDigits
+  - trims number of decimal part specified by maxDecimalDigits
+  - numbers starting or ending with decimal point like .1 or 1. are considered
+    as valid input
+ */
+GenericValidator {
+    id: root
+
+    property string decimalPoint: "."
+    property int maxIntegralDigits: 10
+    property int maxDecimalDigits: 10
+
+    validate: {
+        if (input.length === 0)
+            return GenericValidator.Intermediate
+
+        const charSetRegex = root.maxDecimalDigits > 0 ? /^[0-9\.\,]*$/
+                                                       : /^[0-9]*$/
+        const validCharSet = charSetRegex.test(input)
+
+        if (!validCharSet)
+            return GenericValidator.Invalid
+
+        const delocalized = input.replace(/,/g, ".")
```

Comment:
Could there be a case when e.g. in German locale, thousands separator dot (.) and decimals separator comma (,) are both present in input? If yes, then `delocalized` will be incorrect

Reply:
No matter what is pressed "." or "," it's replaced on the fly with `Locale.decimalPoint`. Therefore thousands separators are not supported.

Reply:
And I see that as long as Amount is not "pastable" there are no issues with usecase like 1,000,000.01

Reply:
The user is "pastable" but only when is valid or it can be done valid. So you can paste both "0.234" and "0,234" but for both cases result will be the same, depending on decimalPoint of your locale.